### PR TITLE
Fix unused-parameter warnings, unnecessary unused attributes

### PR DESCRIPTION
**Pull request by norihiro**
### Description

- Add `OBS_UNUSED` attribute to unused function parameters.
- Remove `UNUSED_PARAMETER` from used parameters
- Remove `OBS_UNUSED` from a used parameter

### Motivation and Context

Don't want to see warnings.
Don't want to see unnecessary `UNUSED_PARAMETER`.

### How Has This Been Tested?

- Reviewed code to ensure the changes are just touching `OBS_UNUSED` and `UNUSED_PARAMETER`.
- Checked warnings built on Linux, macOS, and Windows.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/obs-filters/noise-suppress-filter.c [hunk 1]
```diff
@@ -340,8 +340,6 @@ static inline void alloc_channel(struct noise_suppress_data *ng,
 #ifdef LIBSPEEXDSP_ENABLED
 	ng->spx_states[channel] =
 		speex_preprocess_state_init((int)frames, sample_rate);
-#else
-	UNUSED_PARAMETER(sample_rate);
```
Comment (tytan652) [inline plugins/obs-filters/noise-suppress-filter.c:344]: The related function definition should not have `OBS_UNUSED` since it is used if speexdsp is enabled. So it would be better to remove the `OBS_UNUSED` and keep the `UNUSED_PARAMETER.`

Comment (tytan652) [reply plugins/obs-filters/noise-suppress-filter.c]: Don't forget to mark this "review" as resolved.

**Comment by jp9000** [comment]
You don't need to mark unused parameters in function definitions in C++. You can just remove the parameter's name. (Can't do that in C though, yet)

**Review by RytoEX — CHANGES_REQUESTED** [review]
The change to remove variables in the first commit I would expect to be on its own.

The fourth commit has a change/revert to obs-filters which should be applied to the third commit to revert the change there.

Otherwise, seems fine.

**Comment by norihiro** [comment]
I have updated as Jim and RytoEX commented.
- Unused parameters are removed from `OBSBasic::LoadTransitions`.
- Separate the commit to remove `ms = ticks * AUDIO_OUTPUT_FRAMES * 1000 / sample_rate;` into standalone commit.
- The commits to `plugins/obs-filters/noise-suppress-filter.c` are reordered so that there is no change-then-revert in the series of commits.

File: UI/window-basic-main-transitions.cpp [hunk 2]
```diff
@@ -1770,8 +1770,7 @@ obs_data_array_t *OBSBasic::SaveTransitions()
 	return transitions;
 }
 
-void OBSBasic::LoadTransitions(obs_data_array_t *transitions,
-			       obs_load_source_cb cb, void *private_data)
```
Comment (RytoEX) [inline UI/window-basic-main-transitions.cpp:1774]: I feel like this change is out of scope for this PR, as we're changing a function signature, even though it's private. I'd be fine with this and its related changes as its own PR.

**Review by RytoEX — CHANGES_REQUESTED** [review]
I think I'd prefer the function signature change with `OBSBasic::LoadTransition` to be its own PR.

**Comment by norihiro** [comment]
After rebasing, I saw some more unused parameter warnings. I will update this PR.

**Comment by norihiro** [comment]
Added these changes after Ryan's review comment.
- File `libobs/util/bmem.c` function `base_set_allocator`: Added `OBS_UNUSED`.
- File `plugins/linux-capture/xcomposite-input.c` function `silence_x11_errors`: Added `OBS_UNUSED`.
- File `plugins/vlc-video/vlc-video-source.c` function `init_screencast_capture`: Removed a local variable `aux`, which is not set, not read at all.
- File `plugins/vlc-video/vlc-video-source.c` function `vlcs_video_format`: Removed two local variables `new_width` and `new_height`, which are initialized to 0 but not set, not read.

**Review by RytoEX — APPROVED** [review]
CI built on all jobs. Visually, this seems fine to me. This is probably fine to merge. Will circle back around to this to merge if no one raises any concerns.

**Comment by tytan652** [comment]
On macOS (or with Clang), `LoadTransitions` emits unused-parameter warning.

```diff
diff --git a/UI/window-basic-main-transitions.cpp b/UI/window-basic-main-transitions.cpp
index 629506eb557d..de2492f68bad 100644
--- a/UI/window-basic-main-transitions.cpp
+++ b/UI/window-basic-main-transitions.cpp
@@ -1771,7 +1771,8 @@ obs_data_array_t *OBSBasic::SaveTransitions()
 }
 
 void OBSBasic::LoadTransitions(obs_data_array_t *transitions,
-			       obs_load_source_cb cb, void *private_data)
+			       OBS_UNUSED obs_load_source_cb cb,
+			       OBS_UNUSED void *private_data)
 {
 	size_t count = obs_data_array_count(transitions);
 

```

**Comment by norihiro** [comment]
> On macOS (or with Clang), `LoadTransitions` seems to emit unused-parameter warning.

Thank you for reviewing this PR. The other PR (#5148) was just now merged and it fixed `LoadTransitions`.
